### ISSUE TITLE
git commit is needlessly noisy when there aren't files to adjust

### ISSUE THREAD

**Issue by JasonGross** [coq/coq#12049]
#### Description of the problem

```
$ git commit -a
Auto fixing whitespace and formatting issues...
Patches are saved in 'git-fix-ws-index.XvjzzK', 'git-fix-ws-index-fixed.1Tnb8h' and 'git-fix-ws-tree.ta3FQk'.
If an error destroys your changes you can recover using them.
(The files are cleaned up on success.)

Fixing staged changes...
ocamlformat: Must specify at least one input file, or `-` for stdin

Fixing unstaged changes...
ocamlformat: Must specify at least one input file, or `-` for stdin

Whitespace and formatting pass complete.
```

At least get rid of ```ocamlformat: Must specify at least one input file, or `-` for stdin``` by manually checking for the file list being empty.

#### Coq Version

b26d1f477990d88e235ffda0f23f494456ce5862

**Comment by Zimmi48** [comment]
I agree we should at least get rid of the `ocamlformat: Must specify at least one input file, or `-` for stdin` error. As for the verbosity of the rest, I don't know. @SkySkimmer seemed to think that was fine, while @ppedrot complained about it on Gitter. Maybe there's a way to reduce it a little.

The `ocamlformat: Must specify at least one input file, or `-` for stdin` error only appears when no ML files are updated and you have `ocamlformat` in PATH. Similarly, when some ML files are updated and you have the wrong version of `ocamlformat` in PATH, you can get errors like:

```
ocamlformat: Error while parsing .ocamlformat:
             For option "version": expecting "0.13.0" but got "0.14.0"
```

Since the output of `ocamlformat` is rarely interesting anyway, it seems to me that the best way to fix these two issues would be to redirect the output to `/dev/null`. I'll open a PR to do this in a minute.

**Comment by SkySkimmer** [comment]
>@SkySkimmer seemed to think that was fine

I don't care about commit hook noise because I don't see it. Do not  ask me to judge how important it is because I cannot.

**Comment by Zimmi48** [comment]
Oh, because you call it from Emacs?

**Comment by SkySkimmer** [comment]
>Oh, because you call it from Emacs?

From magit (if I just used shell in emacs I'd see it for instance)

**Comment by ejgallego** [comment]
Same than @SkySkimmer , I never see it unless I need to debug something.
